server/handlers/requestHandlers.go

```diff
@@ -126,15 +130,38 @@ func (h *Handlers) TokenValid(w http.ResponseWriter, r *http.Request) bool {
 	if err != nil {
 		if err == jwt.ErrSignatureInvalid {
 			w.WriteHeader(http.StatusUnauthorized)
-			return false
+			return
 		}
 		w.WriteHeader(http.StatusBadRequest)
-		return false
+		return
 	}
-
 	if !tkn.Valid {
 		w.WriteHeader(http.StatusUnauthorized)
-		return false
+		return
+	}
+
+	if time.Unix(claims.ExpiresAt, 0).Sub(time.Now()) > TIME_BEFORE_EXPIRED {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	expirationTime := time.Now().Add(AUTO_LOGOUT_TIME)
+
+	claims.ExpiresAt = expirationTime.Unix()
+
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
+	tokenString, err := token.SignedString(jwtKey)
+
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
 	}
-	return true
+
+	http.SetCookie(w,
+		&http.Cookie{
+			Name:    "token",
+			Value:   tokenString,
+			Expires: expirationTime,
+		})
+
```

Comment:
Refresh method need to be called by the client side, was trying to see if I can refresh them automatically but online source suggests that it's best practice for client side to refresh jwt token